### ISSUE TITLE
Feature Request: Load pre-trained weights (.pth files etc.) into a flux model

### ISSUE THREAD

**Issue by ejmeitz** [FluxML/Flux.jl#2164]
### Motivation and description

I think it would be useful to load pre-trained weights from PyTorch or Tensorflow. I'm sure this has been discussed before (e.g., in the PyTorch feature parity doc), but I could not an open issue on this.

I work on a Julia package where loading these weights is required. My current work around takes the .pth file and opens it with Pickle.jl. I am still working to parse the resulting dictionary to create a Flux model. It would make my life much easier if there was an associated flux function to just load pre-trained weights and evaluate a model.

**Comment by ToucheSir** [comment]
Are you aware of <https://fluxml.ai/Flux.jl/stable/saving/#Flux.loadmodel!>? I'm not sure it's our place to be including functionality for converting between PyTorch and Flux model structures given the lack of uniformity on the PyTorch side (excepting specific cases like Metalhead.jl where we have a known, limited set of models to map from).

**Comment by ejmeitz** [comment]
Yes, I believe we discussed this on Slack yesterday haha. That is what I plan on using for my implementation as it is only for a single model. Just thought I'd post an issue here since I also saw it in several open threads online and in the Pytorch feature parity document in this repo.

**Comment by ToucheSir** [comment]
I would say it's partially covered by the "We should expose the possibility to load pretrained weights" point under "PyTorch Extras" in https://github.com/FluxML/Flux.jl/issues/1431. As for more general solutions, were someone to come up with a general Dict -> nested struct transformation which works with most PyTorch models, we could consider depending/integrating/advertising it on the Flux side.

**Comment by ejmeitz** [comment]
I'll let you know if what I come up with is general enough.

Flux should have all the same layer types & hyperparameters as PyTorch correct (with different names)?

**Comment by ToucheSir** [comment]
Not necessarily, which is another reason thisi s difficult to generalize. In general we try to keep to close to PyTorch if there's no good reason to diverge, but that's not a hard rule.

**Comment by CarloLucibello** [comment]
Some scripts for porting weights can be found in the Metalhead repo https://github.com/FluxML/Metalhead.jl/tree/master/scripts

**Comment by CarloLucibello** [comment]
Closing — loading the parameters is covered by `Flux.state` + `Flux.loadmodel!`, and Metalhead keeps porting scripts for concrete architectures under https://github.com/FluxML/Metalhead.jl/tree/master/scripts.

A general PyTorch-to-Flux structure converter is not something Flux can offer reliably, since the mapping depends on how each model happens to be laid out; that broader wish stays tracked under #1431.
